src/lib/Controller/api/receiver.c

```diff
@@ -191,145 +191,162 @@ void wb_receiver_init(WbDevice *d) {
 
 void wb_receiver_enable(WbDeviceTag tag, int sampling_period) {
   if (sampling_period < 0) {
-    fprintf(stderr, "Error: wb_receiver_enable() called with negative sampling period.\n");
+    fprintf(stderr, "Error: %s() called with negative sampling period.\n", __FUNCTION__);
     return;
   }
-
   robot_mutex_lock_step();
   Receiver *rs = receiver_get_struct(tag);
-  if (rs) {
+  if (!rs)
+    fprintf(stderr, "Error: %s(): invalid device tag.\n", __FUNCTION__);
+  else {
     rs->enable = true;
     rs->sampling_period = sampling_period;
-  } else
-    fprintf(stderr, "Error: wb_receiver_enable(): invalid device tag.\n");
+  }
```

Comment:
What is the advantage of inverting the condition here?

Reply:
In general, it is always better to rule out bad cases (errors) before addressing the normal case. This is also more consistent with similar tests in this file.

Reply:
> In general, it is always better to rule out bad cases (errors) before addressing the normal case.

This is clearly true if the negative implies a return, in other cases that's more arguable, but this is a detail.